File: php72/crud/app/Repositories/UserRepo.php
<?php

namespace App\Repositories;

use Illuminate\Http\Response;
use App\Models\User;

/**
 * Class ApiController
 *
 * @package App\Http\Controllers
 * @version 1.0
 * @author Angelo (LED) <angelolandino@htomail.it>
 * @copyright Copyright (c) 2015-present
 */
class UserRepo
{
    function __construct(User $user)
    {
        $this->model = $user;
    }

    public function getAll($filters = []): array
    {
        return $this->model->get($filters)->toArray();
    }

    public function whereRaw($filters = []): array
    {
        return $this->model->whereRaw($filters)->get()->toArray();
    }

    public function store(array $data): array
    {
        return $this->model->create($data)->toArray();
    }
}

File: golang/app/repositories/UserRepo.go
package repositories

import (
	"app/app/models"
	"gopkg.in/mgo.v2/bson"
)

type UserRepo struct {
	model models.User
}

func (u *UserRepo) Init() {
	u.model = models.User{}
}

func (u *UserRepo) GetAllUserHasHeatZone(filters bson.M) []models.User {
	models, _ := u.model.Get(filters)
	return models
}

func (u *UserRepo) ShowAll(filters bson.M) []models.User {
	models, _ := u.model.Get(filters)
	return models
}

func (u *UserRepo) Store(body bson.M) (bson.M, error) {

	user := models.User{
		User: body["user"].(string),
		X: int(body["x"].(float64)),
		Y: int(body["y"].(float64)),
	}

	u.model.Create(&user)

	return body, nil
}